helix-term/src/commands/lsp.rs

```diff
@@ -1023,8 +1023,9 @@ fn goto_impl(
         [location] => {
             jump_to_location(editor, location, offset_encoding, Action::Replace);
         }
+        // XXX: All call sites pass non empty containers. This pattern is redundant.
```

Comment:
Let's have this be `[] => unreachable!("goto_impl must be passed a non-empty locations vec")` and add a line of documentation for the function that says that `locations` must be non-empty

Reply:
Addressed in https://github.com/helix-editor/helix/pull/9382/commits/e6ba42934868f9cef9d55e2a37534e7f1c30fd31.